src/client/java/teammates/client/scripts/sql/DataMigrationForCourseEntitySql.java

```diff
@@ -148,4 +186,200 @@ private Student createStudent(teammates.storage.sqlentity.Course newCourse,
 
         return newStudent;
     }
+
+    @Override
+    @SuppressWarnings("unchecked")
+    protected void doOperation() {
+        log("Running " + getClass().getSimpleName() + "...");
+        log("Preview: " + isPreview());
+        setMigrationCriteria();
+
+        Cursor cursor = readPositionOfCursorFromFile().orElse(null);
+        if (cursor == null) {
+            log("Start from the beginning");
+        } else {
+            log("Start from cursor position: " + cursor.toUrlSafe());
+        }
+
+        boolean shouldContinue = true;
+        while (shouldContinue) {
+            shouldContinue = false;
+            Query<Course> filterQueryKeys = getFilterQuery().limit(BATCH_SIZE);
+            if (cursor != null) {
+                filterQueryKeys = filterQueryKeys.startAt(cursor);
+            }
+
+            QueryResults<?> iterator = filterQueryKeys.iterator();
+
+            Course currentCourse = null;
```

Comment:
Could we standardize the naming with the chain PRs (i.e `newCourse` and `oldCourse`)